samples/property/consumer/src/main.rs

```diff
@@ -29,71 +27,42 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     // Obtain the DTDL for the ambient air temmpterature.
     info!("Sending a find_by_id request for entity id {} to the In-Vehicle Digital Twin Service URI {IN_VEHICLE_DIGITAL_TWIN_SERVICE_URI}",
         sdv::vehicle::cabin::hvac::ambient_air_temperature::ID);
     let mut client = DigitalTwinClient::connect(IN_VEHICLE_DIGITAL_TWIN_SERVICE_URI).await?;
     let request = tonic::Request::new(FindByIdRequest {
-        entity_id: String::from(sdv::vehicle::cabin::hvac::ambient_air_temperature::ID),
+        id: String::from(sdv::vehicle::cabin::hvac::ambient_air_temperature::ID),
     });
     let response = client.find_by_id(request).await?;
-    let dtdl = response.into_inner().dtdl;
-    debug!("Received the response for the find_by_id request.");
-
-    debug!("Parsing the DTDL.");
-    let mut parser = ModelParser::new();
-    let json_texts = vec![dtdl];
-    let model_dict_result = parser.parse(&json_texts);
-    if let Err(error) = model_dict_result {
-        panic!("Failed to parse the DTDL: {error}");
-    }
-    let model_dict = model_dict_result.unwrap();
-    debug!("The DTDL parser has successfully parsed the DTDL.");
-
-    // Create the id (as a DTMI) for the ambient air temperature property.
-    let ambient_air_temperature_property_id: Option<Dtmi> =
-        create_dtmi(sdv::vehicle::cabin::hvac::ambient_air_temperature::ID);
-    if ambient_air_temperature_property_id.is_none() {
-        panic!("Unable to create the dtmi");
-    }
-
-    // Get the entity from the DTDL for the ambient air temperature property.
-    let entity_result = model_dict.get(&ambient_air_temperature_property_id.unwrap());
-    if entity_result.is_none() {
-        panic!("Unable to find the entity");
-    }
-    let entity = entity_result.unwrap();
-
-    // Get the URI property from the entity.
-    let uri_property_result = entity.undefined_properties().get(sdv::property::uri::ID);
-    if uri_property_result.is_none() {
-        panic!("Unable to find the URI property");
-    }
-    let uri_property = uri_property_result.unwrap();
-
-    // Get the value for the URI property.
-    let uri_property_value_result = uri_property.get("@value");
-    if uri_property_value_result.is_none() {
-        panic!("Unable to find the value for the URI for ambient air temperature's provider.");
-    }
-    let uri_property_value = uri_property_value_result.unwrap();
-    let uri_str_option = uri_property_value.as_str();
-    let uri = String::from(uri_str_option.unwrap());
-    info!("The URI for the ambient air temperature's provider is {uri}");
-
-    let consumer_uri = format!("http://{CONSUMER_ADDR}"); // Devskim: ignore DS137138
+    let response_inner = response.into_inner();
+    debug!("Received the response for the find_by_id request");
+    info!("response_payload: {:?}", response_inner.entity_access_info);
+
+    let provider_uri = match response_inner.entity_access_info {
```

Comment:
consider using expect here as suggested previously

also, the code for all of the sample consumers appears to be very similar. sample providers are the same way. is there a way we can condense them or abstract out the common functionality so that we don't have duplicate or near duplicate code?

Reply:
I want each sample to be a standalone piece.  Happy if there is redundancy.